cli/admin/src/main/java/io/pravega/cli/admin/dataRecovery/RecoverFromStorageCommand.java

```diff
@@ -0,0 +1,595 @@
+        Services.startAsync(debugStreamSegmentContainer, executorService).get(TIMEOUT.toMillis(), TimeUnit.MILLISECONDS);
+        return debugStreamSegmentContainer;
+    }
+
+    /**
+     * A comparator to sort the segment chunks based on Epoch first and
+     * then the offset.
+     */
+    private static class FileComparator implements Comparator<File>, Serializable {
+        @Override
+        public int compare(File f1, File f2) {
+            String[] file1 = f1.getName().split(EPOCH_SPLITTER);
+            String[] file2 = f2.getName().split(EPOCH_SPLITTER);
+            String file1Epoch = file1[file1.length - 1].split("-")[0];
+            String file2Epoch = file2[file2.length - 1].split("-")[0];
+            if (Long.parseLong(file1Epoch) != Long.parseLong(file2Epoch)) {
+                return Long.valueOf(file1Epoch).compareTo(Long.valueOf(file2Epoch));
+            }
+            file1 = f1.getName().split(OFFSET_SPLITTER);
+            file2 = f2.getName().split(OFFSET_SPLITTER);
+            return Long.valueOf(file1[file1.length - 1].split("\\.")[0]).compareTo(Long.valueOf(file2[file2.length - 1].split("\\.")[0]));
+        }
+    }
+
+    @Override
+    public void execute() throws Exception {
+        ensureArgCount(2);
+        String tableSegmentDataChunksPath = getArg(0);
+        String container = getArg(1);
+
+        if (!NumberUtils.isNumber(container)) {
+            Preconditions.checkArgument(container.toLowerCase().equals("all"), "Container argument should either be ALL/all or a container id.");
+        }
+        @Cleanup
+        Context context = createContext(executorService);
+        @Cleanup
+        val zkClient = createZKClient();
+        val bkConfig = getCommandArgs().getState().getConfigBuilder().include(BookKeeperConfig.builder().with(BookKeeperConfig.ZK_ADDRESS, getServiceConfig().getZkURL())).build().getConfig(BookKeeperConfig::builder);
+        @Cleanup
+        val dataLogFactory = new BookKeeperLogFactory(bkConfig, zkClient, executorService);
+        output("Container Count = %d", this.containerCount);
+        dataLogFactory.initialize();
+        output("Started ZK Client at %s.", getServiceConfig().getZkURL());
+        output("Starting recovery...");
+        // STEP 1: Get all the segment chunks related to the main Segment in order.
+        File[] allFiles = new File(tableSegmentDataChunksPath).listFiles();
+
+        int containerId = NumberUtils.isNumber(container) ? Integer.parseInt(container) : 0;
+        int endContainer = containerId + 1;
+        if (container.equalsIgnoreCase(ALL_CONTAINERS)) {
+            containerId = 0;
+            endContainer = getServiceConfig().getContainerCount();
+        }
+
+        while ( containerId < endContainer ) {
+            recoverFromStorage(containerId, allFiles, context, dataLogFactory);
```

Comment:
Not suggesting we should do it right away. 
But I'm just curious whether it makes sense to run this in parallel for each container to speedup the recovery time.

Reply:
yes..this is in TODO list.

Reply:
yes. This is in TODO list.